src/SDiagonal.jl

```diff
@@ -0,0 +1,106 @@
+# Originally contributed by D. Getz (https://github.com/getzdan), M. Schauer
```

Comment:
When I do this kind of thing, I'd be inclined to add the original chunk of code (in probably non-working form) under the other author's name as a single commit (`@mention`ing them to make sure they're happy with that). Then add any necessary changes under your own name in further commits.

The nice thing about doing it this way is you avoid baking authorship into comments which people feel they can't remove (like the one above), but you also get to do proper attribution which I feel is quite important.

Just some thoughts, I'm happy this was merged already.